examples_tests/43.AllocatorTest/main.cpp

```diff
@@ -1,53 +1,219 @@
 #define _IRR_STATIC_LIB_
-
 #include <irrlicht.h>
-#include "irr/core/alloc/GeneralpurposeAddressAllocator.h"
-
-using namespace irr;
+#include <random>
+#include <cmath>
 
+using namespace irr;
+using namespace core;
 
-/*
-	Problems with GeneralPurposeallocator:
-		1. GeneralPurposeallocator is not able to allocate as much address space as it is expected to
-		2. Program doesn't compile, when I try to call safe_shrink_size
-		3. "Assertion failed: false, file *\include\irr\core\alloc\GeneralpurposeAddressAllocator.h, line 368"
+/*
+	If traits say address allocator supports arbitrary order free,
+	then ETAT_CHOOSE_RANDOMLY is valid. Otherwise ETAT_CHOOSE_MOST_RECENTLY_ALLOCATED
 */
-
-int main()
+
+enum E_TRAITS_ALLOCATION_TYPE
+{
+	ETAT_CHOOSE_RANDOMLY,
+	ETAT_CHOOSE_MOST_RECENTLY_ALLOCATED,
+	ETAT_COUNT
+};
+
+struct AllocationCreationParameters
+{	
+	size_t multiAllocCnt;              //! Specifies amount of adress to be allocated with certain choosen allocator
+	size_t adressesToDeallocateCnt;    //! Specifies amount of adress to be deallocated with certain choosen allocator. Must be less than all allocated and must pick number less than `traits::max_multi_free`, but we don't have `max_multi_free`
```

Comment:
the number of multi_allocs and how many addresses each multi_alloc should allocate should be random.